learning/step_by_step/godot_design_philosophy.rst

```diff
@@ -0,0 +1,145 @@
+Godot’s design philosophy
+=========================
+
+**Every game engine is different and fits different needs**. They do not
+only offer a wide range features: each engine’s design is unique. This
+leads to different workflows and different ways to reason about your
+games’ structure. This all stems from their design philosophy.
+
+This page is here to **help you understand how Godot works**, starting
+from some of its core pillars. It is not a list of available features or
+an engine comparison. To know whether any engine can be a good fit for
+your project or not, you need to both try it out for yourself and
+understand its design and limitations.
+
+Object-oriented design
+----------------------
+
+Godot embraces object-oriented design at its core with its **flexible
+scene system and Node hierarchy**. It tries to stay away from strict
+programming patterns to offer an intuitive way to structure your game.
+
+A Godot scene could be a Weapon, a Character, an Item, a Door, a Level,
+part of a level… anything you’d like. It works like a class in pure code
+except you’re free to design your scene using the editor, using only the
+code, or mixing and matching the two.
+
+It’s different from prefabs you find in several 3d engines as you can
+inherit from and extend scenes. For example you may create a Magician
+that extends your Character scene. Edit the Character and the Magician
+will update as well. It helps you build your projects so that their
+structure match the game’s design.
+
+|image0|
+
+Also note that Godot offers many different types of objects called
+nodes, each with a specific purpose. Nodes are part of a tree and always
+inherit from their parents up to the Node class. Although the engine
+does feature some components like collision shapes, they’re the
+exception, not the norm.
+
+|image1|
+
+Sprite is a Node2D, a CanvasItem and a Node. It has all the properties
+and features of its three parent classes, like transforms or the ability
+to draw custom shapes and render with a custom shader.
+
+All-inclusive package
+---------------------
+
+Godot tries to provide its own tools to answer all of the most common
+needs. It has a dedicated scripting workspace, an animation editor, a
+tilemap editor, a shader editor, etc.
+
+|image2|
+
+The goal is to offer a full package to create games and a continuous
+user experience. You can still work with external programs as long as
+there is an import plugin for it or you can create one, like the `Tiled
+map editor importer <https://github.com/vnen/godot-tiled-importer>`__.
+
+That is also partly why Godot offers its own programming languages
+GDscript and VisualScript, along with C#. They’re designed for the needs
+of game developers, game designers, and they’re tightly integrated in
+the engine and the editor.
```

Comment:
Suggestion (or smth along these lines):

This allows Godot (and GDScript) to combine the fast development speed typical for dynamic languages with some of the niceties usually allowed by a more static type system: the Godot editor can infer things like events and classes used in the code and offer better completion and assistance.